### PR TITLE
Reorganize and cleanup proto definitions.

### LINKED ISSUE #1537: Organize and simplify proto files
The proto files in GraphScope have not been organized for quite a long time while the GraphScope has changed a lot. It could be
- organized
- simplified for remove no longer used parts
- refactored and unified for common utilities
- heavily commented in order for outside collaborators to catch up easily

And the directory of proto can be unified, to eliminate the `proto/graphscope/proto` directory.

### PR DESCRIPTION
**Pull request by siyuan0322**
This PR does the following stuffs:

- [x] move *.proto into `python/graphscope/proto`, and change proto generating related stuff correspondingly.
- [x] To make proto interface simple and understandable, remove or unify some unused fields (may not be concise and correct, subject to change):
    - [x] rm place_holder
    - [x] rename func
    - [x] modify_type, report_type, type seems not appropriate in value
    - [x] EngineService -> AnalyticalOnly
    - [ ] ~~unify two datatype pb (invovles broader stuff, another PR later)~~
    - [x] rm fetch
    - [x] simplify OpResults
    - [x] mv query_args to types.proto

~~Major changes: Flatten attr_value? make it less abstract, so that it's more easy to be understand.~~
- [x] Add rich comments

Fixes #1537

### CODE REVIEW

File: python/graphscope/proto/op_def.proto [hunk 1]
```diff
@@ -64,28 +57,17 @@ message OpResult {
   // if success, store the metrics. (e.g. how many seconds used, memory bytes...)
   string metrics = 3;
```
Comment (yecol) [inline python/graphscope/proto/op_def.proto:58]: I suggest we wrap an object-like meta, including:
-  status-code?/error-code?
- details: string additional info?
- metrics: time/memory 
- has_large_results(?)

**Review by yecol — CHANGES_REQUESTED** [review]

File: python/graphscope/proto/engine_service.proto [hunk 2]
```diff
@@ -19,7 +19,7 @@ package gs.rpc;
 
 import "graphscope/proto/message.proto";
 
-service EngineService {
+service GAEService {
```
[the remark this answers is not in the document]
Comment (siyuan0322) [reply python/graphscope/proto/engine_service.proto:22]: Wants to be specific. I'm thinking `engine` there is not specific and maybe confusing for those who don't know the history of GAE. And given we have incorporated part of GIE and GLE into the DAG.

File: python/graphscope/proto/op_def.proto (hunk 1 above)
Comment (siyuan0322) [reply python/graphscope/proto/op_def.proto:58]: Good suggestion~

File: python/graphscope/proto/op_def.proto [hunk 3]
```diff
@@ -49,9 +45,6 @@ message OpDef {
   // arguments that served as application querying parameters.
   // Such as source vertex id for SSSP.
   QueryArgs query_args = 8;
```
[the remark this answers is not in the document]
Comment (siyuan0322) [reply python/graphscope/proto/op_def.proto:47]: This query_args will wrap specific data type message in data_types.proto, in order to unpack arguments with correct types in runtime in app_frame.cc, app_invoker.cc, by meta-template programming.

It's not very trivial to use attr_value to replace this.

**Review by yecol — APPROVED** [review]

**Review by sighingnow — APPROVED** [review]

**Pull request merged**